sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -365,6 +367,8 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
     val cols = Option(ctx.identifierList()).map(visitIdentifierList).getOrElse(Nil)
     val partitionKeys = Option(ctx.partitionSpec).map(visitPartitionSpec).getOrElse(Map.empty)
 
+    blockBang(ctx.errorCapturingNot())
```

Comment:
does it work if we override `def visitErrorCapturingNot` and put the check there?

Reply:
I tried that. But I need to invoke visitErrorCapturingNot explicitly which makes it worse.

Reply:
ah I see